Storage/Processor/Processor.php

```diff
@@ -219,14 +219,19 @@ public function getResource(Config $config, string $filename)
             return $this->storageManager->getResource($config->getAssetHash(), $config->getStorageContext());
         }
 
-        try {
-            return $this->storageManager->getResource($config->getAssetHash(), $config->getConfigHash());
-        } catch (NotFoundException $e) {
-            $generatedResource = $this->generateResource($config);
-            if ($cacheableResult) {
-                $this->storageManager->cacheResource($generatedResource, $config->getAssetHash(), $config->getConfigHash(), $filename, $config->getMimeType());
+        $cacheableResult = $config->cacheableResult();
+        if ($cacheableResult && !$noCache) {
+            try {
+                return $this->storageManager->getResource($config->getAssetHash(), $config->getConfigHash());
+            } catch (NotFoundException $e) {
+                //we have tried to get a cached version matching this request
             }
-            return $generatedResource;
         }
+
+        $generatedResource = $this->generateResource($config);
+        if ($cacheableResult) {
+            $this->storageManager->cacheResource($generatedResource, $config->getAssetHash(), $config->getConfigHash(), $filename, $config->getMimeType());
```

Comment:
What if the storage is not available and caching fails? Should we not try catch it to prevent error?

Reply:
I added a try catch with a warning log